File: Assets/Script/bezier.js
////using UnityEngine;
////using System.Collections.Generic;
////[RequireComponent(typeof (LineRenderer))]
//
//var controlPoints = new Array();
////    public Transform[] controlPoints;
//var LineRenderer: lineRenderer;
//
//private
//var curveCount: int = 0;
//private
//var layerOrder: int = 0;
//private
//var SEGMENT_COUNT: int = 50;
//
//function start() {
//    //    if (!lineRenderer) {
//    //        lineRenderer = GetComponent < LineRenderer > ();
//    //    }
//    lineRenderer.sortingLayerID = layerOrder;
//    //    curveCount = (int) controlPoints.Length / 3;
//    curveCount = Mathf.FloorToInt(controlPoints.Length / 3);
//}
//
//function Update() {
//    DrawCurve();
//}
//
//function DrawCurve() {
//    for (var j = 0; j < curveCount; j++) {
//        for (var i = 1; i <= SEGMENT_COUNT; i++) {
//            //            var t: float = i / (float) SEGMENT_COUNT;
//            var t: float = i / SEGMENT_COUNT;
//            var nodeIndex: int = j * 3;
//            var pixel: Vector3 = CalculateCubicBezierPoint(t, controlPoints[nodeIndex].position, controlPoints[nodeIndex + 1].position, controlPoints[nodeIndex + 2].position, controlPoints[nodeIndex + 3].position);
//            lineRenderer.SetVertexCount(((j * SEGMENT_COUNT) + i));
//            lineRenderer.SetPosition((j * SEGMENT_COUNT) + (i - 1), pixel);
//        }
//
//    }
//}
//
//function CalculateCubicBezierPoint(t: float, p0: Vector3, p1: Vector3, p2: Vector3, p3: Vector3) {
//    var u: float = 1 - t;
//    var tt: float = t * t;
//    var uu: float = u * u;
//    var uuu: float = uu * u;
//    var ttt: float = tt * t;
//
//    var p: Vector3 = uuu * p0;
//    p += 3 * uu * t * p1;
//    p += 3 * u * tt * p2;
//    p += ttt * p3;
//
//    return p;
//}

File: Assets/Script/Bezier.cs
using UnityEngine;
using System.Collections.Generic;
[RequireComponent(typeof(LineRenderer))]
public class Bezier : MonoBehaviour
{
       public Renderer rend;
    public Transform[] controlPoints;
    public LineRenderer lineRenderer;

    private int curveCount = 0;
    private int layerOrder = 0;
    private int SEGMENT_COUNT = 50;
     public float linepapa=0.99993f;


    void Start()
    {
        if (!lineRenderer)
        {
            lineRenderer = GetComponent<LineRenderer>();
        }
        lineRenderer.sortingLayerID = layerOrder;
        print(controlPoints.Length);
        curveCount = (int)controlPoints.Length / 3;

               rend = GetComponent<Renderer>();
//        rend.material.shader = Shader.Find("SoftParticlesFactor");
    }

    void Update()
    {
        Color shininess = new Color(1.0F, 1.0F, 1.0F,1-linepapa);
      rend.material.SetColor("_TintColor", shininess);
        DrawCurve();
    }

    void DrawCurve()
    {
        for (int j = 0; j <curveCount; j++)
        {
            for (int i = 1; i <= SEGMENT_COUNT; i++)
            {
                float t = i / (float)SEGMENT_COUNT;

                linepapa= Time.time% 1;

                if(t>linepapa){
                    t=linepapa;
                }

                int nodeIndex = j * 3;
                Vector3 pixel = CalculateCubicBezierPoint(t, controlPoints [nodeIndex].position, controlPoints [nodeIndex + 1].position, controlPoints [nodeIndex + 2].position, controlPoints [nodeIndex + 3].position);
                lineRenderer.SetVertexCount(((j * SEGMENT_COUNT) + i));
                lineRenderer.SetPosition((j * SEGMENT_COUNT) + (i - 1), pixel);
            }

        }
    }

    Vector3 CalculateCubicBezierPoint(float t, Vector3 p0, Vector3 p1, Vector3 p2, Vector3 p3)
    {
        float u = 1 - t;
        float tt = t * t;
        float uu = u * u;
        float uuu = uu * u;
        float ttt = tt * t;

        Vector3 p = uuu * p0;
        p += 3 * uu * t * p1;
        p += 3 * u * tt * p2;
        p += ttt * p3;

        return p;
    }
}
